src/System.Drawing.Common/src/System/Drawing/Graphics.cs

```diff
@@ -20,6 +21,369 @@ public sealed partial class Graphics : MarshalByRefObject, IDisposable, IDeviceC
+                        value = InterpolationMode.Bilinear;
+                        break;
+                    case InterpolationMode.High:
+                        value = InterpolationMode.HighQualityBicubic;
+                        break;
+                    case InterpolationMode.Invalid:
+                        throw new ArgumentException(SR.GdiplusInvalidParameter);
+                    default:
+                        break;
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetInterpolationMode(new HandleRef(this, NativeGraphics), unchecked((int)value));
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        public bool IsClipEmpty
+        {
+            get
+            {
+                int isEmpty;
+                int status = SafeNativeMethods.Gdip.GdipIsClipEmpty(new HandleRef(this, NativeGraphics), out isEmpty);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return isEmpty != 0;
+            }
+        }
+
+        public bool IsVisibleClipEmpty
+        {
+            get
+            {
+                int isEmpty;
+                int status = SafeNativeMethods.Gdip.GdipIsVisibleClipEmpty(new HandleRef(this, NativeGraphics), out isEmpty);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return isEmpty != 0;
+            }
+        }
+
+        public float PageScale
+        {
+            get
+            {
+                var scale = new float[] { 0.0f };
+                int status = SafeNativeMethods.Gdip.GdipGetPageScale(new HandleRef(this, NativeGraphics), scale);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return scale[0];
+            }
+
+            set
+            {
+                if (value <= 0 || value > 1000000032)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetPageScale(new HandleRef(this, NativeGraphics), value);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        public GraphicsUnit PageUnit
+        {
+            get
+            {
+                int unit = 0;
+                int status = SafeNativeMethods.Gdip.GdipGetPageUnit(new HandleRef(this, NativeGraphics), out unit);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return (GraphicsUnit)unit;
+            }
+            set
+            {
+                if (value < GraphicsUnit.World || value > GraphicsUnit.Millimeter)
+                {
+                    throw new InvalidEnumArgumentException(nameof(value), unchecked((int)value), typeof(GraphicsUnit));
+                }
+
+                if (value == GraphicsUnit.World)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
```

Comment:
Ditto

Reply:
This fixes the test `PageUnit_SetWorld_ThrowsArgumentException` with libgdiplus